### PR TITLE
Add support for creating OL8 AMIs

### LINKED ISSUE #24: [Feature Request] Add Support for OL8
**Is your feature request related to a problem? Please describe.**

With the change from CentOS 8 to CentOS 8 "core" and "stream" and the pending termination of support for "core", some AMIgen users are looking at alternatives to CentOS 8. Need to add support for other options. This request is for [Oracle Linux 8](https://docs.oracle.com/en/operating-systems/oracle-linux/8/).

**Describe the solution you'd like**

Allow AMIgen8 to produce Oracle Linux 8 AMIs

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #24

### CODE REVIEW

**Comment by scoter-oracle** [comment]
There's a comment reporting:

"This kernel not only contains OL8-specific, proprietary extensions,"

please, consider that UEK does not have any kind of proprietary extension and the source-code of the same is available on GitHub at https://github.com/oracle/linux-uek

File: docs/README_OracleLinux.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## DNF Uniqueness
+
+Oracle Linux introduces distribution-specific build-order dependencies that effect the initial population of the build's target-disk not previously encountered by this project. In order to work past these dependencies and succesfully produce an OL8 image, it is necessary to invoke the `OSpackages.sh` script with the `--setup-dnf` variable-flag. This flag must be passed a comma-delimited list of variable-names and variable-values. For access to Oracle's public repositories, the recommended valie for this flag is `ociregion=,ocidomain=oracle.com`. Failure to set this flag will result in the `OSpackages.sh` script failing during baseline population of the build process's target-disk with errors similar to:
```
Comment (lorengordon) [inline docs/README_OracleLinux.md:11]:
```suggestion
Oracle Linux introduces distribution-specific build-order dependencies that effect the initial population of the build's target-disk not previously encountered by this project. In order to work past these dependencies and successfully produce an OL8 image, it is necessary to invoke the `OSpackages.sh` script with the `--setup-dnf` variable-flag. This flag must be passed a comma-delimited list of variable-names and variable-values. For access to Oracle's public repositories, the recommended value for this flag is `ociregion=,ocidomain=oracle.com`. Failure to set this flag will result in the `OSpackages.sh` script failing during baseline population of the build process's target-disk with errors similar to:
```

File: docs/README_OracleLinux.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+If using the scripts' default mode, Oracle Linux will be installed using the `@core` RPM-group. Oracle, unlike Red Hat or the various Enterprise Linux distros, defines a customized kernel (the [Unbreakable Enterprise Kernel](https://docs.oracle.com/en/operating-systems/uek/), or "UEK") as its default kernel. While this kernel is customized with notionally [open extensions](https://github.com/oracle/linux-uek), the baseline kernel used for adding those extensions is a different fork-point off the Linux kernel-project than the one used by Red Hat's distro &hellip;or the other distros that seek to be downstream clones of Red Hat's distro: Red Hat based it's kernel on [4.18.0](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/8.0_release_notes/index#kernel); Oracle bases theirs on [5.4.17](https://docs.oracle.com/en/operating-systems/uek/). To minimize kernel-differences between the various Enterprise Linux distros this project publishes, this default is overridden. If readers of this document are using this project's contents and do not desire the use of the UEK kernel, invoke the `OSpackages.sh` with the flag and option, `-x kernel-uek`
```
Comment (lorengordon) [inline docs/README_OracleLinux.md:7]: kinda some weird character in the updated text...

```suggestion
If using the scripts' default mode, Oracle Linux will be installed using the `@core` RPM-group. Oracle, unlike Red Hat or the various Enterprise Linux distros, defines a customized kernel (the [Unbreakable Enterprise Kernel](https://docs.oracle.com/en/operating-systems/uek/), or "UEK") as its default kernel. While this kernel is customized with notionally [open extensions](https://github.com/oracle/linux-uek), the baseline kernel used for adding those extensions is a different fork-point off the Linux kernel-project than the one used by Red Hat's distro, the other distros that seek to be downstream clones of Red Hat's distro: Red Hat based it's kernel on [4.18.0](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/8.0_release_notes/index#kernel); Oracle bases theirs on [5.4.17](https://docs.oracle.com/en/operating-systems/uek/). To minimize kernel-differences between the various Enterprise Linux distros this project publishes, this default is overridden. If readers of this document are using this project's contents and do not desire the use of the UEK kernel, invoke the `OSpackages.sh` with the flag and option, `-x kernel-uek`
```

Comment (ferricoxide) [reply docs/README_OracleLinux.md:7]: It's an HTML entity for a horizontal ellipsis ("…"). While GitHub markdown will typically render the directly-typed ellipsis character, there can be some portability issues if the entity isn't used (get similar behavior with the – litereal vice &ndash;).

File: OSpackages.sh [hunk 3]
```diff
@@ -199,6 +210,24 @@ function PrepChroot {
       ln -t "${CHROOTMNT}/etc" -s ./rc.d/init.d
    fi
 
+   # Satisfy weird, OL8-dependecy:
+   # * Ensure the /etc/dnf and /etc/yum contents are present
+   if [[ ${#DNF_ARRAY[*]:-} -gt 0 ]]
```
Comment (lorengordon) [inline OSpackages.sh:215]: This actually results in a "bad substitution" error. Sorry about that. This suggestion works though:

```suggestion
   if [[ -n "${DNF_ARRAY:-}" ]]
```

**Review by lorengordon — APPROVED** [review]
build passed 🚢

**Pull request merged**